### PR TITLE
GHC 8.4 support (continued)

### PR DESCRIPTION
**Pull request by purcell**
This builds on top of https://github.com/fimad/prometheus-haskell/pull/36, and applies the changes requested there which @zyla hadn't yet found time for.

### CODE REVIEW

**Comment by purcell** [comment]
Ugh - one build task is currently timing out for unrelated reasons, but the other two are 👍 .

**Comment by purcell** [comment]
I re-pushed the changes to trigger a fresh build, and it all looks clear this time. :-)

**Comment by jml** [comment]
Thanks! I’ll try to take a look before Monday.

**Comment by jml** [comment]
That did not happen. Still want to get to this soon.

**Comment by purcell** [comment]
Cool, thanks.

File: prometheus-metrics-ghc/src/Prometheus/Metric/GHC.hs [hunk 1]
```diff
@@ -125,13 +134,193 @@ ghcCollectors = [
...
+            CounterType
+            (rtsTimeToSeconds . elapsed_ns)
+
+    , statsCollector
+            "ghc_gcdetails_gen"
+            "The generation number of this GC"
+            HistogramType -- TODO: is this correct?
+                          -- Gauge makes little sense here.
+                          -- With Histogram we'll be able to see which
+                          -- generations are collected most often.
```
Comment (jml) [inline prometheus-metrics-ghc/src/Prometheus/Metric/GHC.hs]: I don't want to create stop energy, so feel free to disregard this and proceed with merging.

On reflection, we should probably not be exporting this at all, at least, not as it is.

If I understand correctly (a big "if"), there should be a labelled Counter, where the label is the generation number, and the value is the number of GCs for that generation.

Further, the current generation should be applied as a label to things like `ghc_gcdetails_copied_bytes`, which would make the metrics more useful.

Realistically, I think it's fine to release these with the metrics clearly marked as alpha & subject to change without prior deprecation warnings in future releases. Nothing will beat actually using these.

Comment (purcell) [reply prometheus-metrics-ghc/src/Prometheus/Metric/GHC.hs]: Yep, agree with this - fine to push this out and then change it later IMO.

**Comment by purcell** [comment]
Would it be a big job to make a fresh release containing these changes? Let me know if I can help with anything.

**Comment by ocharles** [comment]
I think it's going to get rolled knot the 1.0 changes. I plan to release
this week.

**Comment by purcell** [comment]
Awesome - thanks Olly!

**Comment by purcell-nec** [comment]
> I think it's going to get rolled knot the 1.0 changes. I plan to release this week.

Cheeky ping - let me know if I can help with anything.

**Comment by ocharles** [comment]
Ah darn, I'm about to go on holiday. Next Friday ok?

**Comment by purcell** [comment]
Yup, sure. Just double-checking it's still on the radar: aiming to get an app upgraded to GHC 8.4, and this is the last remaining hurdle. Enjoy your holiday!

**Pull request merged**